### DISCUSSION TITLE
xonsh history stdout show

### DISCUSSION THREAD

**Discussion by rickysarraf** [xonsh/xonsh#5613] [Q&A]
I have `xonsh` configured to have command `stdout` to be stored into the history backend. All that seems to be working fine.

```
♅ 21:27:32 rrs@xps ~/r/d/ap/apt-offline    master
@ $XONSH_CAPTURE_ALWAYS
True
⛢ 21:32:12 rrs@xps ~/r/d/ap/apt-offline    master
@ $XONSH_STORE_STDOUT
True
⛢ 21:32:17 rrs@xps ~/r/d/ap/apt-offline    master
@ date
Saturday 20 July 2024 09:32:32 PM IST
♒ 21:32:32 rrs@xps ~/r/d/ap/apt-offline    master
@ __xonsh__.history[-1].out
'Saturday 20 July 2024 09:32:32 PM IST\n'
ॐ 21:32:39 rrs@xps ~/r/d/ap/apt-offline    master
```

```
@ history info
backend: sqlite
sessionid: dbf5ad63-f288-4cdf-8cea-23364e511fd2
filename: /home/rrs/.local/share/xonsh/xonsh-history.sqlite
session items: 6
all items: 4554
gc options: (100000, 'commands')
```

My question is that how do I get the same ` __xonsh__.history[-1].out` content via the `history` command. Something like `history show --with-stdout id`

**Comment by anki-code** [comment]
I'm not the user of json history (i'm using sqlite where I can extract anything using SQL lang) so if you can't find this may be it's new feature.

The start point to implement is here - https://github.com/xonsh/xonsh/tree/32a085543a7594a39fa61062cbaa06cc540f23fc/xonsh/history

**Reply by rickysarraf** [reply]
Thank you. Yes. I'm using `sqlitebrowser` and it does a good job. But as a user, it was my next obvious expectation to see something within `history` itself.

Now that it is clear that that functionality isn't there yet, it is a good motivation to implement. Thank you. 🙏🏽

**Reply by anki-code** [reply]
`history` alias is here https://github.com/xonsh/xonsh/blob/32a085543a7594a39fa61062cbaa06cc540f23fc/xonsh/history/main.py#L224

e.g. `history info` - https://github.com/xonsh/xonsh/blob/32a085543a7594a39fa61062cbaa06cc540f23fc/xonsh/history/main.py#L402
